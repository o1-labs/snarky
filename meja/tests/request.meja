request('a) Request(list('a));

request('a) Request2(list('a))
  with handler (l) => { Snarky__Request.Repond (List.hd(l)); };
