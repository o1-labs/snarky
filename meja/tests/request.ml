module Impl = Snarky.Snark.Make (Snarky.Backends.Mnt4.Default)
open Impl

include struct
  type _ Snarky.Request.t += Request : 'x list -> 'x Snarky.Request.t
end

include struct
  type _ Snarky.Request.t += Request2 : 'x Snarky.Request.t

  let handle_Request2 = function
    | With {request= Request2; respond} ->
        let unhandled = Snarky.Request.unhandled in
        let _ = () in
        unhandled
    | _ ->
        Snarky.Request.unhandled
end

include struct
  type _ Snarky.Request.t += Request3 : 'x option -> 'x Snarky.Request.t

  let handle_Request3 = function
    | With {request= Request3 x; respond} -> (
        let unhandled = Snarky.Request.unhandled in
        match x with
        | None ->
            unhandled
        | Some x ->
            respond (Snarky.Request.Response.Provide x) )
    | _ ->
        Snarky.Request.unhandled
end
