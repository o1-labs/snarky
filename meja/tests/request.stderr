File "tests/request.meja", line 4, characters 24-46:
Error: Unbound constructor Snarky__Request.Repond.
